CMakeLists.txt

```diff
@@ -88,13 +88,28 @@ endif()
 
 find_package(Python3 ${LLVM_MINIMUM_PYTHON_VERSION} REQUIRED COMPONENTS Interpreter Development)
 
+# On MacOS, find_package for Java and JNI doesn't work if JAVA_INCLUDE_PATH
+# and JAVA_INCLUDE_PATH2 are not set. Setting them requires JAVA_HOME to be
+# known. If the JAVA_HOME env var isn't set, we try to figure it out by running
+# java and looking for the java.home property.
+if(APPLE)
+  set(JAVA_HOME $ENV{JAVA_HOME})
+  if ("${JAVA_HOME}" STREQUAL "")
+    execute_process(
+      COMMAND bash -c "java -XshowSettings:properties -version 2>&1 | \
+                       sed -n 's/[ \t]*java.home = //p'"
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+      OUTPUT_VARIABLE JAVA_HOME)
+  endif()
+  if (NOT "${JAVA_HOME}" STREQUAL "")
+    SET(JAVA_INCLUDE_PATH  "${JAVA_HOME}/include")
+    SET(JAVA_INCLUDE_PATH2 "${JAVA_HOME}/include/darwin")
+  endif()
```

Comment:
Should we issue a warning when its not found?
Also, same question as asked on another MacOS issue: is the setting of JAVA_HOME documented?

Reply:
find_package will do that.